### PR TITLE
added security schema info into the operation generation model

### LINKED ISSUE #2055: SwaggerToCSharpControllerGenerator does not support security definitions
When I generate my controllers there are no "Authorize" attributes being added.

I'm using a swagger 2.0 spec with a security definition that looks like this (globally):

```yaml
securityDefinitions:
  bearerAuth:
    type: apiKey
    in: header
    name: Authorization
    description: my description
security:
  - bearerAuth: []
```

I've had to use my own liquid template to fix the routeprefix issue (#1386), and I was hoping I could solve this security definitions issue myself as well but unfortunately the [CSharpOperationModel](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs#L19) doesn't provide any of the required data to the template.

Perhaps the [CSharpOperationModel](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs#L19) should provide a `public bool RequiresAuthentication { get; set; }` property that can be used in-order to add the `[Authorize]` attribute.

### PR DESCRIPTION
**Pull request by Place1**
this should fix #2055

i've only added the data to the code generation model to avoid a breaking change to the default template.

### CODE REVIEW

File: src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs [hunk 1]
```diff
@@ -206,6 +206,12 @@ public string RouteName
             }
         }
 
+        /// <summary>True if the operation has any security schemes</summary>
+        public bool RequiresAuthentication => _operation.ActualSecurity.Count() != 0;
+
+        /// <summary>Gets the security schemas that apply to this operation</summary>
+        public IEnumerable<OpenApiSecurityRequirement> ActualSecurity => _operation.ActualSecurity;
```
Comment (RicoSuter) [inline src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs]: Let's name this property just Security

File: src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs [hunk 2]
```diff
@@ -206,6 +206,12 @@ public string RouteName
             }
         }
 
+        /// <summary>True if the operation has any security schemes</summary>
+        public bool RequiresAuthentication => _operation.ActualSecurity.Count() != 0;
```
Comment (RicoSuter) [inline src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs]: Is ActualSecurity never null?

Comment (Place1) [reply src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs]: nice catch!

File: src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs (hunk 1 above)
Comment (Place1) [reply src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs]: sure thing

**Comment by uniquelau** [comment]
@Place1 could you share an example of the liquid template you are using?

I've had a little play around, and I can see, the `Operation.ActualSecurity` is being set, but I can't figure out how to loop over the values.

**Comment by RicoSuter** [comment]
The property name in the template is "Security".

**Comment by uniquelau** [comment]
@RicoSuter thanks - would I be right in thinking that I need to extend the class in some way to be able to get access to the required data? I've had a little play inside `Controller.Method.Annotations.liquid`.

```
{% if operation.RequiresAuthentication -%}
    {% for o in operation.Security -%}
        [Authorize({{ o }})]
    {% endfor -%}
{% endif %}
```

```
Results in:

[Authorize([bearerAuth, System.Collections.Generic.List`1[System.Object]])]
```

**Comment by RicoSuter** [comment]
o is also a complex object

**Comment by RicoSuter** [comment]
Not sure whether its a good idea to directly expose the openapi classes - maybe we should map them to template models.

**Comment by RicoSuter** [comment]
Sorry o is a dictionary:

https://apiverse.io/organizations/nuget-org/nuget-feeds/public/packages/NSwag.Core/versions/latest?type=NSwag.OpenApiSecurityRequirement&namespace=All&assembly=netstandard-2.0

Its hard to work with dicts in the tpls

**Pull request merged**